.github/workflows/ci.yml

```diff
@@ -7,20 +7,57 @@ on:
   pull_request:
 
 jobs:
+  check-integrity:
+    name: Check integrity
+    runs-on: ubuntu-latest
```

Comment:
Using the latest version of Ubuntu can lead two unexpected problems:
1. It may happen when the latest version changes to the newer Ubuntu, the Github actions will fail because the version of tool set changes.
2. You can fix the issue 1. on master but it may happen you have an old branch with the "old" Ubuntu that will not work properly because it has not been updated with the latest master.

Therefore, I RECOMMEND to speify the ubuntu version to ubuntu-22.04. [More info](https://github.com/actions/runner-images)

Reply:
Good idea, but this was actually a copy from the `build` job in this workflow. I guess I should fix that one as well.